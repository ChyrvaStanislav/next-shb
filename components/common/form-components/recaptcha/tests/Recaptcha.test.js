import React from 'react';
import { render } from '@testing-library/react';
import Recaptcha from '../index';

const setFieldValue = jest.fn();

const Component = () => (
  <Recaptcha
    formId="formId"
    label="Some label"
    field={{
      name: 'name',
      value: '',
    }}
    form={{
      touched: {},
      errors: {},
      setFieldValue
    }}
  />
);

const OLD_ENV = process.env;

beforeEach(() => {
  jest.resetModules(); // Most important - it clears the cache
  process.env = { ...OLD_ENV }; // Make a copy
});

afterAll(() => {
  process.env = OLD_ENV; // Restore old environment
});

describe('Recaptcha renders', () => {
  it('should Recaptcha renders', () => {
    process.env.RECAPTCHA_KEY = '6Leta8IUAAAAAGc5AiKzXuv7qJaQjL_nM3rSDYpN';
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
