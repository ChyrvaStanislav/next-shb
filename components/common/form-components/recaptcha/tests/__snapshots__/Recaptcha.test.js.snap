// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recaptcha renders should Recaptcha renders 1`] = `
<DocumentFragment>
  <section
    class="textareaContainer"
  >
    <label
      class="small_filters label"
      for="formId-name"
      other=""
    >
      Some label
    </label>
    <div
      data-testid="recaptcha"
      render="explicit"
    />
  </section>
</DocumentFragment>
`;
